cuqi/distribution/_smoothed_laplace.py

```diff
@@ -0,0 +1,80 @@
+import numpy as np
+from cuqi.distribution import Distribution
+
+class SmoothedLaplace(Distribution):
+    """ Smoothed Laplace distribution. 
+
+    Defines a smoothed Laplace distribution given a location, a scale and a smoothing parameter
+    beta. The smoothed Laplace distribution is defined as
+
+    .. math::
+
+        p(x) = \\frac{1}{2b} \exp\left(-\\frac{\sqrt{(x-\mu)^2 + \beta}}{b}\\right),
+
+    where :math:`\mu` is the location (mean), :math:`b` is the scale (decay) parameter and
+    :math:`\beta` is the smoothing parameter.
+
+    The rate parameter is defined as :math:`\lambda = \\frac{1}{b}`.
+
+    The variables of this Laplace distribution are independent identically distributed (i.i.d.).
+
+    Parameters
+    ----------
+    location : scalar, list, tuple, or ndarray
+        The location parameter of the distribution.
+
+    scale : scalar, list, tuple, or ndarray
+        The scale parameter of the distribution.
+    
+    beta : scalar
+
+    """
+
+    def __init__(self, location, scale, beta=1e-3, **kwargs):
```

Comment:
Can you ensure the variables location and scale can be set to None? This is needed for conditioning, Gibbs etc.
It can be achieved using getters and setters. See e.g. Cauchy.py for an example that is somewhat similar.
The mutable variables for the distribution here should return location, scale.

Reply:
Also add defaults location=None, scale=None please :)

Reply:
Done as suggested. Thanks!